scripts/weekly-summary

```diff
@@ -272,6 +269,72 @@ def print_test_details(scenario, days, test_name, buf):
                 print(f'        {start_time} ({elapsed_time}s): {msg}', file=buf)
 
 
+def kstest_logdir(tmpdir, test):
+    """
+    Return the directory of the logs for the test
+
+    The logfile path should start with /var/tmp/kstest-*
+    but this finds the kstest-* no matter what the leading
+    path elements are.
+    """
+    logfile = test["logfile"]
+    for e in logfile.split(os.path.sep):
+        if e.startswith("kstest-"):
+            return os.path.join(tmpdir, e)
+
+    raise RuntimeError(f"No kstest-* directory found in {logfile}")
+
+
+def archive_test_logs(days, archive_path, all_logs):
+    """
+    Store the logs from the failed/flaky tests for archiving
+    and later examination. Paths follow the pattern:
+
+    MM-DD-YYYY / scenario / failed|flakes / TESTNAME / [1...] /
+
+    and each directory contains the logs for that run of the test.
+    Currently this is the kstest.log and virt-install.log files
+    """
+    for day in days.keys():
+        daydir = datetime.strptime(day, "%Y%m%d").strftime("%m-%d-%Y")
+        for scenario in days[day].keys():
+            # temporary log directories are stored by scenario + date
+            datename = scenario + "-" + datetime.strptime(day, "%Y%m%d").strftime("%Y-%m-%d")
+            if datename not in all_logs:
+                raise RuntimeError(f"Missing all_log entry for {datename}")
+
+            if not os.path.exists(all_logs[datename].name):
+                raise RuntimeError(f"Missing log directory for {datename}")
```

Comment:
Is this actually a problem? Sometimes the tests stop working due to infra outages, and then a partial summary might still be useful.

I'll admit that I don't remember the rest of the code involved, so I can't say if that's a problem for any other reason.

Reply:
This is here because it's possible to only run the report using the daily .json summaries. If you do that then the logs aren't present and it cannot include them. Right now we are running it with the full archive, and regenerating the daily json summaries so it isn't going to hit this.